hipparchus-core/src/main/java/org/hipparchus/fraction/BigFraction.java

```diff
@@ -338,23 +339,19 @@ public BigFraction(final long num, final long den) {
         this(BigInteger.valueOf(num), BigInteger.valueOf(den));
     }
 
+    @FunctionalInterface
+    public interface ConvergenceTest {
+        boolean test(long numerator, long denominator);
+    }
+
     /** Generate a {@link Stream stream} of convergents from a real number.
      * @param value value to approximate
      * @param maxConbvergents maximum number of convergents.
      * @return stream of {@link BigFraction} convergents approximating  {@code value}
      * @since 2.1
      */
-    public static Stream<BigFraction> convergents(final double value, final int maxConvergents) {
```

Comment:
Could we just keep both the existing `convergents` method which provides all intermediates in addition to the new `convergent` method which avoids building these intermediate terms? They seem to answer different use cases to me.
Of course, the constructors above that only need the last convergent should use the new method as you propose.

I see in the tests that it is possible to use `convergent` to keep the intermediate by adding them as a side effect of checking convergence, but it seems awkward to me, so I would prefer to have both methods.

What do you think?

Reply:
> 
> What do you think?

Fully agree. I didn't liked the side-effects as well.